### ISSUE TITLE
Dialog a11y

### ISSUE THREAD

**Issue by Heydon** [iTwin/stratakit#1231]
## 1. Implement expected behavior

Implements cyclical focus (quasi focus trap) rather than expected/native behavior. User cannot currently reach browser chrome (address bar etc) when dialog is open.

**Comment by mayank99** [comment]
Documenting my findings for posterity…

If we use [`disableEnforceFocus`](https://mui.com/material-ui/api/modal/#modal-prop-disableEnforceFocus) to remove the focus trap, we'd need to manually `inert` everything outside the `Dialog`. This is not straightforward.

I looked into two approaches:

1. If we assume that the [root portal container](https://stratakit.bentley.com/docs/components/root/#portal-container) is always a direct child of `<body>`, then we could attach a shadowRoot to `document.body`. Inside it, the default unnamed slot would become `inert`. A separate named slot would be used for the root portal container.
    ```html
    <body>
      <template shadowrootmode="open">
        <slot inert></slot>
        <slot name="portal-container"></slot>
      </template>
    </body>
    ```
   This makes most of the page content inert, but any portaled elements outside the Dialog would still be focusable, and to make it worse, they would have `aria-hidden` added to them (by MUI).

2. We could find the Dialog's farthest ancestor and add an attribute to it, e.g. `data-modal`. Then, we could use [`interactivity`](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/interactivity) to make everything else inert.

    ```css
    :has(> [data-modal]) > *:not([data-modal]) {
      interactivity: inert;
    }
    ```

   Or more directly, look for [`aria-modal`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-modal) across the entire page (might be less performant):

    ```css
    body:has([role="dialog"][aria-modal="true"]) {
      > *:not(:has([role="dialog"][aria-modal="true"])) {
        interactivity: inert;
      }
    }
    ```

   Either way, the experimental nature of `interactivity` and its poor support makes this approach unreliable (only useful as progressive enhancement).

The first approach could have worked better if the `Dialog` was a direct child of `<body>` (no longer the case after #1513).

**Comment by GerardasB** [comment]
- [ ] `aria-hidden` applied on siblings of portalled element might cause browser to output some errors when portalling is implemented. See https://github.com/iTwin/stratakit/pull/1762#discussion_r3825919674, https://github.com/iTwin/stratakit/pull/1506#discussion_r3645310736
  ```
  Blocked aria-hidden on an element because its descendant retained focus. The focus must not be hidden from assistive technology users. Avoid using aria-hidden on a focused element or its ancestor. Consider using the inert attribute instead, which will also prevent focus. For more details, see the aria-hidden section of the WAI-ARIA specification at https://w3c.github.io/aria/#aria-hidden.
  Element with focus: <button.MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-sizeMedium MuiButton-colorSecondary mui-1pk93v4-MuiButtonBase-root-MuiButton-root>
  Ancestor with aria-hidden: <div.MuiDialogActions-root MuiDialogActions-spacing mui-15fu35s-MuiDialogActions-root> 
  ```

**Comment by mayank99** [comment]
> `aria-hidden` applied on siblings of portalled element

This was fixed upstream at [material-ui#49026](https://redirect.github.com/mui/material-ui/pull/49026). Need to re-test it when released.
